contracts/Lock.sol

```diff
@@ -117,25 +123,52 @@ contract Lock is AragonApp, IForwarder, IForwarderFee {
     /**
     * @notice Locks the required amount of tokens and executes the specified action
     * @dev IForwarder interface conformance. Consider using pretransaction on UI for necessary approval.
+    *      We check for edge cases where an approve is made and before this function is called, one of msg.sender current locks is unlocked resulting in a different griefing amount.
     * @param _evmCallScript Script to execute
     */
-    function forward(bytes _evmCallScript) public {
+    function forward(bytes _evmCallScript) public  auth(LOCK_TOKENS_ROLE) {
         require(canForward(msg.sender, _evmCallScript), ERROR_CAN_NOT_FORWARD);
 
+        (uint256 griefAmount, uint256 griefDuration) = getGriefing(msg.sender);
+
+        uint256 allowance = token.allowance(msg.sender, address(this));
+        uint256 computedAmount = lockAmount.add(griefAmount);
+
+        uint256 totalAmount = allowance > computedAmount ? allowance : computedAmount;
+        uint256 totalDuration = lockDuration.add(griefDuration);
+
         WithdrawLockLib.WithdrawLock[] storage addressWithdrawLocks = addressesWithdrawLocks[msg.sender];
-        uint256 unlockTime = getTimestamp().add(lockDuration);
-        addressWithdrawLocks.push(WithdrawLockLib.WithdrawLock(unlockTime, lockAmount));
+        uint256 unlockTime = getTimestamp().add(totalDuration);
+        addressWithdrawLocks.push(WithdrawLockLib.WithdrawLock(unlockTime, totalAmount));
 
-        require(token.safeTransferFrom(msg.sender, address(this), lockAmount), ERROR_TRANSFER_REVERTED);
+        require(token.safeTransferFrom(msg.sender, address(this), totalAmount), ERROR_TRANSFER_REVERTED);
 
-        emit NewLock(msg.sender, unlockTime, lockAmount);
+        emit NewLock(msg.sender, unlockTime, totalAmount);
         runScript(_evmCallScript, new bytes(0), new address[](0));
     }
 
     function getWithdrawLocksCount(address _lockAddress) public view returns (uint256) {
         return addressesWithdrawLocks[_lockAddress].length;
     }
 
+    /**
+    * @notice Get's amount and duration penalty based on the number of current locks `_sender` has
+    * @param _sender account that is going to lock tokens
+    * @return amount and duration penalty
+    */
+    function getGriefing(address _sender) public view returns (uint256, uint256) {
+        WithdrawLockLib.WithdrawLock[] memory addressWithdrawLocks = addressesWithdrawLocks[_sender];
+
+        uint256 activeLocks = 0;
+        for (uint256 withdrawLockIndex = 0; withdrawLockIndex < addressWithdrawLocks.length; withdrawLockIndex++) {
+            if (getTimestamp() < addressWithdrawLocks[withdrawLockIndex].unlockTime) {
+                activeLocks += 1;
+            }
```

Comment:
To prevent potential OOG issues, however unlikely, we could just use `addressWithdrawLocks.length` to represent the `activeLocks` as opposed to all those that are unlocked, as the `addressWithdrawLocks` list is deleted from when a `WithdrawLock` has been withdrawn. The user would need to be more proactive in withdrawing their locks though. Alternatively we leave it as it is which could prevent someone with many locks from being able to call `forward` until some time has passed. @lkngtn Do you have any thoughts?

Reply:
Thinking frontend side, if we would like to show the account what's going to be the next griefing amounts, it would be easier to take into account the total number of locks instead of only non withdrawable. If we decide to leave it as it is, to show the account these values would require checking `getGriefing` contract function on an interval basis (pretty similar to the vesting issue we had with redemptions)